### PR TITLE
feat(apple): Add docs for Swift Error description

### PR DESCRIPTION
**Pull request by philipphofmann**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [x] Checked Vercel preview for correctness, including links
- [x] PR was reviewed and approved by any necessary SMEs
- [x] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

Sentry Cocoa 8.7.0 contains improvements for Swift error descriptions. This PR adds the docs for this feature. Please only merge this PR after releasing 8.7.0.

### CODE REVIEW

File: src/platform-includes/capture-error/apple.mdx [hunk 1]
```diff
@@ -62,3 +75,15 @@ extension MyCustomError: CustomNSError {
...
+
+## Capturing Uncaught Exceptions in macOS
+
+By default, macOS applications do not crash whenever an uncaught exception occurs. To enable this with Sentry:
+
+1. Open the application's `Info.plist` file
+2. Search for `Principal class` (the entry is expected to be `NSApplication`)
+3. Replace `NSApplication` with `SentryCrashExceptionApplication`
+
+</PlatformSection>
```
Comment (philipphofmann) [inline src/platform-includes/capture-error/apple.mdx:89]: I moved this existing section down here.

**Review by brustolin — APPROVED** [review]

File: src/platform-includes/capture-error/apple.mdx [hunk 2]
```diff
@@ -19,27 +19,37 @@ SentrySDK.capture(exception: exception)
...
 ```
 
-<PlatformSection supported={["apple.macos"]}>
+### Swift Errors
 
-## Capturing Uncaught Exceptions in macOS
+For Swift Errors conforming to the [Error Protocol](https://developer.apple.com/documentation/swift/error) the SDK
+sends the domain, code and the description of the Swift error.
+Before [sentry-cocoa 8.7.0](https://github.com/getsentry/sentry-cocoa/blob/main/CHANGELOG.md#870) the SDK sends only
+the domain and error code.
```
Comment (shanamatthews) [inline src/platform-includes/capture-error/apple.mdx:27]: Is there a reason for these funny line breaks? It doesn't affect the output, but it's a little annoying

File: src/platform-includes/capture-error/apple.mdx [hunk 3]
```diff
@@ -19,27 +19,37 @@ SentrySDK.capture(exception: exception)
 [SentrySDK captureException:exception];
 ```
 
-<PlatformSection supported={["apple.macos"]}>
+### Swift Errors
 
-## Capturing Uncaught Exceptions in macOS
+For Swift Errors conforming to the [Error Protocol](https://developer.apple.com/documentation/swift/error) the SDK
+sends the domain, code and the description of the Swift error.
+Before [sentry-cocoa 8.7.0](https://github.com/getsentry/sentry-cocoa/blob/main/CHANGELOG.md#870) the SDK sends only
```
Comment (shanamatthews) [inline src/platform-includes/capture-error/apple.mdx:26]:
```suggestion
For older versions of the SDK, prior to [sentry-cocoa 8.7.0](https://github.com/getsentry/sentry-cocoa/blob/main/CHANGELOG.md#870) the SDK only sends
```

File: src/platform-includes/capture-error/apple.mdx [hunk 4]
```diff
@@ -19,27 +19,37 @@ SentrySDK.capture(exception: exception)
...
+    case wrongUser(id: String)
+    case wrongPassword
+}
 
-1. Open the application's `Info.plist` file
-2. Search for `Principal class` (the entry is expected to be `NSApplication`)
-3. Replace `NSApplication` with `SentryCrashExceptionApplication`
+SentrySDK.capture(error: LoginError.wrongUser("12345678"))
+```
+For the Swift error above Sentry displays:
```
Comment (shanamatthews) [inline src/platform-includes/capture-error/apple.mdx:37]:
```suggestion
For the Swift error above Sentry would display the following, based on SDK version:
```

File: src/platform-includes/capture-error/apple.mdx [hunk 5]
```diff
@@ -19,27 +19,37 @@ SentrySDK.capture(exception: exception)
...
+| Since 8.7.0 | `LoginError` | `wrongUser(id: "12345678") (Code: 1)` |
+| Before 8.7.0 | `LoginError` | `Code: 1` |
 
-## Customizing Error Descriptions
+### Customizing Error Descriptions
 
 This feature is available on [sentry-cocoa 7.25.0](https://github.com/getsentry/sentry-cocoa/blob/master/CHANGELOG.md#7250) and above.
 
-Sentry will display the error code in the error description field by default. For custom error types, you may want to provide a custom description to make it easier to identify the error in the _Issues_ page. For `NSError` values, this can be done by adding a description to the `userInfo` dictionary with the key `NSDebugDescriptionErrorKey`.
+You may want to provide a custom description to make identifying the error in the _Issues_ page easier. For `NSError` values, this can be done by adding a description to the `userInfo` dictionary with the key `NSDebugDescriptionErrorKey`.
```
Comment (shanamatthews) [inline src/platform-includes/capture-error/apple.mdx:48]:
```suggestion
You may want to provide a custom description to make identifying the error in the **Issues** page easier. For `NSError` values, you can do this by adding a description to the `userInfo` dictionary with the key `NSDebugDescriptionErrorKey`.
```

bold page titles according to style guide

**Review by shanamatthews — APPROVED** [review]
Looks good, just some small tweaks

File: src/platform-includes/capture-error/apple.mdx (hunk 2 above)
Comment (philipphofmann) [reply src/platform-includes/capture-error/apple.mdx:27]: No, not a reason. I removed them.

**Comment by philipphofmann** [comment]
Thanks a lot for the review @shanamatthews 👏

**Pull request merged**